cli/src/commands/check-command.ts

```diff
@@ -1,15 +1,18 @@
 import type { Network } from "../lib";
 import { compareCurrentStateWith } from "../lib";
 import { withSpinner } from "../lib/with-spinner.js";
+import { GithubClient } from "../lib/github-client.js";
 
 export async function checkCommand(
   etherscanKey: string,
   network: Network,
-  upgradeDirectory: string
+  upgradeDirectory: string,
+  ref: string
 ): Promise<void> {
   const { diff, l1Abis } = await withSpinner(
     () => compareCurrentStateWith(etherscanKey, network, upgradeDirectory),
     "Gathering contract data"
   );
-  console.log(await diff.toCliReport(l1Abis, upgradeDirectory));
+  const github = new GithubClient();
+  console.log(await diff.toCliReport(l1Abis, upgradeDirectory, github, ref));
```

Comment:
Now to print the report we need to compare  get the current hashes of the contract to be able to compare.

Reading again this code, I don't love this design. I should refactor this to make `toCliReport` a sync method again.